trace.go

```diff
@@ -108,6 +108,20 @@ func removeObserverFrom(parent **spanObserverTuple, ref *spanObserverTuple) (
 // Id returns the id of the Trace
 func (t *Trace) Id() int64 { return t.id }
 
+// GetAll returns values associated with a trace. See SetAll.
+func (t *Trace) GetAll() (val map[interface{}]interface{}) {
+	t.mtx.Lock()
+	defer t.mtx.Unlock()
+	return t.vals
```

Comment:
We should probably make a copy here, otherwise mutations to the trace it came from will apply to the result here.